modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/import/ImportModal.js

```diff
@@ -12,85 +12,63 @@
  * details.
  */
 
-import ClayButton from '@clayui/button';
-import ClayForm from '@clayui/form';
 import ClayModal from '@clayui/modal';
-import ClayProgressBar from '@clayui/progress-bar';
 import PropTypes from 'prop-types';
 import React from 'react';
 
-import {importStatus} from '../BatchPlannerImport';
-import Poller from '../Poller';
+import ImportPreviewModal from './ImportPreviewModal';
+import ImportProcessModal from './ImportProcessModal';
 
 const ImportModal = ({
 	closeModal,
+	dbFields,
+	fieldsSelections,
+	fileContent,
+	fileFields,
 	formDataQuerySelector,
 	formSubmitURL,
 	observer,
+	portletNamespace,
+	setFileContent,
+	setStartImport,
+	startImport,
 }) => {
-	const {errorMessage, loading, percentage} = Poller(
-		formDataQuerySelector,
-		formSubmitURL,
-		importStatus
-	);
-
-	return (
-		<ClayModal observer={observer} size="md">
-			<ClayModal.Header>
-				{Liferay.Language.get('import')}
-			</ClayModal.Header>
-
-			<ClayModal.Body>
-				<ClayForm.Group className={errorMessage ? 'has-error' : ''}>
-					<div className="progress-container">
-						<ClayProgressBar
-							value={percentage}
-							warn={!!errorMessage}
-						/>
-					</div>
+	const handleEditCell = (newValue, cellIndex, rowIndex) => {
+		const newRow = fileContent[rowIndex];
+		const newFileContent = fileContent;
+		Object.entries(newRow).forEach(([key], index) => {
+			if (cellIndex === index) {
+				newRow[key] = newValue;
+			}
+		});
 
-					{errorMessage && (
-						<ClayForm.FeedbackGroup>
-							<ClayForm.FeedbackItem>
-								<ClayForm.FeedbackIndicator symbol="exclamation-full" />
+		newFileContent.splice(rowIndex, 1, newRow);
+		setFileContent(newFileContent);
+	};
 
-								{errorMessage}
-							</ClayForm.FeedbackItem>
-						</ClayForm.FeedbackGroup>
-					)}
-				</ClayForm.Group>
-			</ClayModal.Body>
-
-			<ClayModal.Footer
-				last={
-					<ClayButton.Group spaced>
-						<ClayButton
-							displayType="secondary"
-							onClick={closeModal}
-						>
-							{Liferay.Language.get('cancel')}
-						</ClayButton>
-
-						<ClayButton
-							disabled={loading}
-							displayType="primary"
-							onClick={closeModal}
-							type="submit"
-						>
-							{loading && (
-								<span className="inline-item inline-item-before">
-									<span
-										aria-hidden="true"
-										className="loading-animation"
-									></span>
-								</span>
-							)}
+	return (
+		<ClayModal observer={observer} size={startImport ? 'lg' : 'md'}>
+			{!startImport && (
+				<ImportPreviewModal
+					closeModal={closeModal}
+					dbFields={dbFields}
+					fieldsSelections={fieldsSelections}
+					fileContent={fileContent}
```

Comment:
why both `fileContent` and `fileFields`?

Reply:
i need fileContent in order to populate the preview table body

Reply:
my point is that `fileContent` also contains `fileFields` so maybe you don't need to pass both the variables. 
If you're happy with passing both, leave it this way, but keep in mind that there is some duplication that can be avoided